src/recipients/recipient.ts

```diff
@@ -174,26 +175,34 @@ export function getRecipientFromAddress(
 
 export function getRecipientVerificationStatus(
   recipient: Recipient,
-  e164NumberToAddress: E164NumberToAddressType
+  e164NumberToAddress: E164NumberToAddressType,
+  addressToVerificationStatus: AddressToVerificationStatus
 ): RecipientVerificationStatus {
-  if (recipientHasAddress(recipient)) {
-    return RecipientVerificationStatus.VERIFIED
-  }
-
   if (!recipientHasNumber(recipient)) {
-    return RecipientVerificationStatus.UNKNOWN
-  }
+    if (recipientHasAddress(recipient) && recipient.address in addressToVerificationStatus) {
+      switch (addressToVerificationStatus[recipient.address]) {
+        case true:
+          return RecipientVerificationStatus.VERIFIED
+        case false:
+          return RecipientVerificationStatus.UNVERIFIED
```

Comment:
is this backwards compatible? seems like we previously returned VERIFIED for all addresses.

Reply:
yeah, if we're in the old flow, we set the status to verified in the verification hook one level up and don't call this function.